### PR TITLE
Dev/login

### PR DESCRIPTION
**Pull request by bakunyo**
https://github.com/hr-dash/hr-dash/pull/84 を先に

・deviseのroutes設定を変更（参考： https://github.com/hr-dash/hr-dash/issues/81 ）
・deviseのI18n対応（Gem追加）
・ログインして無ければログイン画面に遷移させる
・ログイン関連のテスト追加

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,14 @@
 Rails.application.routes.draw do
 
-  devise_for :users
-  devise_for :admin_users, ActiveAdmin::Devise.config
+  devise_for :users, skip: [:sessions]
+  devise_scope :user do
+    resources :sessions, only: [:new, :create] do
```
Comment (freezer530000) [inline config/routes.rb:5]: **[rails_best_practices]**
- restrict auto-generated routes sessions (only: []) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

File: spec/requests/session_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    describe "redirect this if user hasn't login" do
+      before { get root_path }
+      it { expect(response).to redirect_to(new_session_path) }
+    end
+  end
+
+  describe '#create POST /sessions' do
+    let(:login_params) { { user: { email: user.email, password: user.password } } }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb:15]: **[rubocop]**
- Line is too long. [83/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: config/routes.rb [hunk 3]
```diff
@@ -16,6 +23,8 @@
     end
   end
   root to: 'sample#index'
+
+  resources 'monthly_reports', except: :destroy
```
Comment (freezer530000) [inline config/routes.rb]: **[rails_best_practices]**
- restrict auto-generated routes monthly_reports (only: [:show, :index]) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

File: spec/requests/session_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    before { post sessions_path, login_params }
+
+    context 'valid user' do
+      let(:user) { create :user  }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb]: **[rubocop]**
- Unnecessary spacing detected.

File: spec/requests/session_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+      it { expect(response).to have_http_status :found }
+      it { expect(response).to redirect_to(root_path) }
+      it { expect(controller.current_user).not_to be nil }
+    end
+
+    context 'invalid user' do
+      let(:user) { build :user  }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb]: **[rubocop]**
- Unnecessary spacing detected.

File: spec/requests/session_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+      it { expect(response).to have_http_status :found }
+      it { expect(response).to redirect_to(new_session_path) }
+    end
+  end
+
+  describe '#destroy DELETE /sessions/destroy' do
+    let(:user) { create :user  }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb]: **[rubocop]**
- Unnecessary spacing detected.

File: spec/requests/session_spec.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    let(:login_params) { { user: { email: user.email, password: user.password } } }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb]: **[rubocop]**
- Line is too long. [83/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: spec/requests/session_spec.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      let(:user) { build :user }
+
+      it { expect(response).to have_http_status :found }
+      it { expect(response).to redirect_to(new_session_path) }
+    end
+  end
+
+  describe '#destroy DELETE /sessions/destroy' do
+    let(:user) { create :user }
+    let(:login_params) { { user: { email: user.email, password: user.password } } }
```
Comment (freezer530000) [inline spec/requests/session_spec.rb:36]: **[rubocop]**
- Line is too long. [83/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: config/routes.rb (hunk 1 above)
Comment (bakunyo) [inline config/routes.rb:5]:
```
class SessionsController < Devise::SessionsController
```

に

```
def new
  super
end
```

とか書けばこのアラート解消できそうですが、特に対応不要かと…。

**Comment by Pujyoooo** [comment]
LGTM

**Comment by ta-chibana** [comment]
LGTM

**Pull request merged**